### PR TITLE
support pointefree mutables

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

File: src/kernel.jl [hunk 1]
```diff
@@ -247,7 +247,7 @@ function set_arg!{T}(k::Kernel, idx::Integer, arg::T)
     ref = if isbits(T_aligned)
         Base.RefValue(packed)
     else
-        packed
+        pointer_from_objref(packed)
```
Comment (vchuravy) [inline src/kernel.jl]: Wouldn't it be fine to just say: `Ref(packed)`? and then let the logic for references take over instead of re-implementing it here?

Comment (SimonDanisch) [reply src/kernel.jl]: Yeah, that might just work...

Comment (vchuravy) [reply src/kernel.jl]: Do you want to quickly fix it otherwise I will merge this and tag a point release.

File: test/test_behaviour.jl [hunk 2]
```diff
@@ -272,4 +272,56 @@ let test_struct = "
         @test all(x -> x == 13.5, r)
     end
 end
+
+end
+
+mutable struct MutableParams
```
Comment (vchuravy) [inline test/test_behaviour.jl]: This won't work on 0.5. See test failure on GPU CI...

**Pull request merged**